endaq/calc/stats.py

```diff
@@ -6,6 +6,342 @@
 
 import numpy as np
 import pandas as pd
+from scipy import signal
+import plotly.graph_objects as go
+
+from endaq.plot import rolling_min_max_envelope
+from endaq.calc import filters, integrate, utils, shock
+
+
+def shock_vibe_metrics(
+        df: pd.DataFrame,
+        tukey_percent: float = 0.1,
+        highpass_cutoff: float = None,
+        vel_disp_multiplier: float = 1.0,
+        zero: typing.Literal["start", "mean", "median", None] = "median",
+        include_integration: bool = True,
+        include_pseudo_velocity: bool = True,
+        damp: float = 0.05,
+        init_freq: float = 1.0,
+        bins_per_octave: float = 12,
+        include_resultant: bool = False,
+        display_plots: bool = False,
+) -> pd.DataFrame:
+    """
+    Compute the following shock and vibration metrics for a given time series dataframe
+        - Peak Absolute Acceleration
+        - RMS Acceleration
+        - Peak Absolute Velocity
+        - RMS Velocity
+        - Peak Absolute Displacement
+        - RMS Displacement
+        - Peak Pseudo Velocity
+    :param df: the input dataframe with an index defining the time in seconds or datetime
+    :param tukey_percent: the portion of the time series to apply a Tukey window (a taper that forces beginning and end to 0), default is 0.1
+        - Note that the RMS metrics will only be computed on the portion of time that isn't tapered
+    :param zero: the output quantity driven to zero prior to the calculations
+        `None` does nothing
+        `"start"` forces the first datapoint to 0,
+        `"mean"` chooses ``-np.mean(output)`` & `"median"` (default) chooses
+        ``-np.median(output)``
+    :param highpass_cutoff: the cutoff frequency of a preconditioning highpass
+        filter; if None, no filter is applied. For shock events, it is recommended to set this to None (the default), but it is recommended for vibration.
+    :param vel_disp_multiplier: applies a scale to the velocity and displacement metrics
+        - 386.09 to convert from g to inches (in)
+        - 9806.65 to convert from g to millimeters (mm)
+    :param include_integration: include the calculations of velocity and displacement, default behavior
+    :param include_pseudo_velocity: include the more time consuming calculation of pseudo velocity, default behavior
+    :param damp: the damping coefficient `ζ`, related to the Q-factor by
+        `ζ = 1/(2Q)`; defaults to 0
```

Comment:
Actual default is different.  I would also add some bounds checking to the code, to make sure the damping ratio is valid.

```suggestion
    :param damp: the damping coefficient `ζ`, related to the Q-factor by
        `ζ = 1/(2Q)`; defaults to 0.05
```